concurrent_test.go

```diff
@@ -0,0 +1,98 @@
+package bbolt_test
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	bolt "go.etcd.io/bbolt"
+	"go.etcd.io/bbolt/internal/btesting"
+)
+
+func TestRepeatableRead(t *testing.T) {
+	const (
+		readInterval  = 50 * time.Millisecond
+		writeInterval = 10 * time.Millisecond
+		testDuration  = 30 * time.Second
+	)
+
+	bucket := []byte("data")
+	key := []byte("foo")
+
+	// prepare the db
+	t.Log("Create the db")
+	db := btesting.MustCreateDB(t)
+	err := db.Update(func(tx *bolt.Tx) error {
+		_, err := tx.CreateBucket(bucket)
+		return err
+	})
+	require.NoError(t, err)
+
+	// start readonly transactions
+	stopCh := make(chan struct{}, 1)
+	var wg sync.WaitGroup
+	t.Log("Create readonly transactions")
+	wg.Add(1)
+	go func() {
```

Comment:
would be good to generalize this code to maintain multiple RO transaction started at random/the same times.

Reply:
Part of this might be factoring the reading and writing workers to a separate methods such that they can be reused for similar tests - e.g working on the bigger volume of data.